src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -177,35 +177,42 @@ public static void Initialize()
             throw;
         }
 
-        RegisterInstrumentations(InstrumentationDefinitions.GetAllDefinitions());
-
-        try
+        if (GeneralSettings.Value.ProfilerEnabled)
         {
-            foreach (var payload in _pluginManager.GetAllDefinitionsPayloads())
+            RegisterInstrumentations(InstrumentationDefinitions.GetAllDefinitions());
```

Comment:
Can we rename `RegisterInstrumentations` to `RegisterByteCodeInstrumentations` to make it obvious that source-code instrumentations will still work?